catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/ui/activity/AccessibilityPreferencesActivityTest.java

```diff
@@ -0,0 +1,156 @@
+import static org.catrobat.catroid.ui.BaseSettingsActivity.ACCESS_STARTER_BRICKS;
+
+@RunWith(AndroidJUnit4.class)
+public class AccessibilityPreferencesActivityTest {
+
+	private List<String> settings = new ArrayList<>(Arrays.asList(ACCESS_LARGE_TEXT, ACCESS_HIGH_CONTRAST,
+			ACCESS_ADDITIONAL_ICONS, ACCESS_LARGE_ICONS, ACCESS_HIGH_CONTRAST_ICONS, ACCESS_LARGE_ELEMENT_SPACING,
+			ACCESS_STARTER_BRICKS, ACCESS_DRAGNDROP_DELAY));
+
+	@Rule
+	public BaseActivityInstrumentationRule<AccessibilityPreferencesActivity> baseActivityTestRule = new
+			BaseActivityInstrumentationRule<>(AccessibilityPreferencesActivity.class, true, false);
+
+	@Before
+	public void setUp() throws Exception {
+		baseActivityTestRule.launchActivity(null);
+	}
+
+	@After
+	public void tearDown() {
+	}
+
+	@Test
+	public void settingsTest() {
+		SharedPreferences sharedPreferences = PreferenceManager
+				.getDefaultSharedPreferences(InstrumentationRegistry.getTargetContext());
+
+		onView(withText(R.string.preference_access_title_fontface)).perform(click());
+		onView(withText(R.string.preference_access_title_fontface_dyslexic)).perform(click());
+		assertEquals(sharedPreferences.getString(ACCESS_MYPROFILE_FONTFACE, ""), ACCESS_FONTFACE_VALUE_DYSLEXIC);
+
+		onView(withText(R.string.close)).perform(click());
+
+		onView(withText(R.string.preference_access_title_large_text)).perform(scrollTo(), click());
+		assertTrue(sharedPreferences.getBoolean(ACCESS_MYPROFILE_LARGE_TEXT, false));
+
+		onView(withText(R.string.preference_access_title_high_contrast)).perform(scrollTo(), click());
+		assertTrue(sharedPreferences.getBoolean(ACCESS_MYPROFILE_HIGH_CONTRAST, false));
+
+		onView(withText(R.string.preference_access_title_additional_icons)).perform(scrollTo(), click());
+		assertTrue(sharedPreferences.getBoolean(ACCESS_MYPROFILE_ADDITIONAL_ICONS, false));
+
+		onView(withText(R.string.preference_access_title_large_icons)).perform(scrollTo(), click());
+		assertTrue(sharedPreferences.getBoolean(ACCESS_MYPROFILE_LARGE_ICONS, false));
+
+		onView(withText(R.string.preference_access_title_high_contrast_icons)).perform(scrollTo(), click());
+		assertTrue(sharedPreferences.getBoolean(ACCESS_MYPROFILE_HIGH_CONTRAST_ICONS, false));
+
+		onView(withText(R.string.preference_access_title_largeelementspacing)).perform(scrollTo(), click());
+		assertTrue(sharedPreferences.getBoolean(ACCESS_MYPROFILE_LARGE_ELEMENT_SPACING, false));
+
+		onView(withText(R.string.preference_access_title_starter_bricks)).perform(scrollTo(), click());
+		assertTrue(sharedPreferences.getBoolean(ACCESS_MYPROFILE_STARTER_BRICKS, false));
+
+		onView(withText(R.string.preference_access_title_dragndrop_delay)).perform(scrollTo(), click());
+		assertTrue(sharedPreferences.getBoolean(ACCESS_MYPROFILE_DRAGNDROP_DELAY, false));
+
+		onView(withId(R.id.scratch_project_apply_button)).perform(click());
+		onView(withText(R.string.ok)).perform(click());
```

Comment:
dont do this. restarting the app and putting it back into main menu activity is going to be a bitch and break other tests. (due to tha background threads, etc.)

the one time i couldnt get around doing something like this (LanguageSwitchMainMenuTest) i had to:
- get the respawned MainMenuActivity and register its idling resource
- write a new testRule that prevents defaultProject beeing generated
to make this somehow work without breaking other tests.

please get updated with develop to enable git hooked jenkins runs for espresso, and run all espresso tests in a bunch to see if new tests break other tests.

Reply:
I would argue that the restart is an essential part of the Accessibility settings, and that it makes sense to test if it correctly quits to the MainMenu, and then check if the UI was correctly adjusted in the settings. Only when i apply the settings (and the app restarts as a result) do the settings get saved in the SharedPreferences. I guess I could call the function to save the settings manually in the test, but that I'd say that defeats the purpose of a UI Test a little bit.

I think it would be worth to go the extra mile, and do what you did to get around breaking it. Do you disagree?

Reply:
ok, then, have a look at the LanguageSwitchMainMenuTest in develop,
- register the idling resource of MainMenuActivity right after you restart
- using the no-DefaultProject rule helps heaps

keep an eye on other tests when running them in conjunction with this one here, and if something else starts randomly breaking, try to remember that this test could be a reason for it.

i probably should add another category annotation for those kinds of tests so one can easily filter for them and see if everything else suddenly works again when not running those. so the set of tests one has to check if that happens would be highlighted...

Reply:
Registered the resource, and applied the no-DefaultProject rule.

"please get updated with develop to enable git hooked jenkins runs for espresso, and run all espresso tests in a bunch to see if new tests break other tests."

I'm afraid I can't do that because this is on the feature/NOLB branch which has a bazillion conflicts with the develop branch, and I honestly don't think i can piece together a working version with my limited knowledge of what does what.

Reply:
"I'm afraid I can't do that because this is on the feature/NOLB branch which has a bazillion conflicts with the develop branch, and I honestly don't think i can piece together a working version with my limited knowledge of what does what."

ok, thats beyond this PR, but well have to sit together and find a way on how to achieve this, rather sooner than later, because its not going to become easier the more code goes in there.

Reply:
i just saw that this seems to have just happened in  https://github.com/Catrobat/Catroid/pull/2564, hence the merge conflicts...

Reply:
Oh great! I guess we will wait until it's merged, to rebase and test this PR?
Speaking of which, how do you run the espresso tests? I only found a "PullRequest-Espresso" Job on Jenkins but it says "This project is currently disabled".

Reply:
I just realized that I already rebased #2564 into this PR. Woops.